templates/srpm.erb

```diff
@@ -0,0 +1,233 @@
+# Hello packaging friend!
+
+# Allow building noarch packages that contain binaries
+%define _binaries_in_noarch_packages_terminate_build 0
+
+<% (attributes[:rpm_filter_from_requires] or []).each do |reqfilter| -%>
+%filter_from_requires <%= reqfilter %>
+<% end -%>
+<% (attributes[:rpm_filter_from_provides] or []).each do |provfilter| -%>
+%filter_from_provides <%= provfilter %>
+<% end -%>
+<% if !(attributes[:rpm_filter_from_requires] or []).empty? or !(attributes[:rpm_filter_from_provides] or []).empty?-%>
+%filter_setup
+<% end -%>
+
+Name: <%= name %>
+Version: <%= version %>
+<% if epoch -%>
+Epoch: <%= epoch %>
+<% end -%>
+Release: <%= iteration or 1 %><%= "%{?dist}" if attributes[:rpm_dist] %>
+<%# use the first line of the description as the summary -%>
+Summary: <%= summary %>
+<% if !attributes[:rpm_autoreqprov?] -%>
+AutoReqProv: no
+<% else -%>
+AutoReqProv: yes
+<% end -%>
+<% if attributes[:rpm_autoreq?] -%>
+AutoReq: yes
+<% end -%>
+<% if attributes[:rpm_autoprov?] -%>
+AutoProv: yes
+<% end -%>
+# Seems specifying BuildRoot is required on older rpmbuild (like on CentOS 5)
+# fpm passes '--define buildroot ...' on the commandline, so just reuse that.
+BuildRoot: %buildroot
+<% if !prefix.nil? and !prefix.empty? %>
+Prefix: <%= prefix %>
+<% end -%>
+
+Group: <%= category %>
+<%# Sometimes the 'license' field has multiple lines... Hack around it. 
+  # While technically yes this means we are 'modifying' the license,
+  # since the job of FPM is to get shit done and that this is  only
+  # modifying whitespace, it should be reasonably OK. -%>
+License: <%= license.gsub("\n", " ") %>
+<% if !vendor.nil? and !vendor.empty? -%>
+Vendor: <%= vendor %>
+<% end -%>
+<% if !url.nil? and !url.empty? -%>
+URL: <%= url %>
+<%else -%>
+URL: http://nourlgiven.example.com/
+<% end -%>
+Packager: <%= maintainer %>
+
+Source0: <%= source_archive %>
+
+<% if !attributes[:no_depends?] -%>
+<% dependencies.each do |req| -%>
+Requires: <%= req %>
+<% end -%>
+<% (attributes[:rpm_tag] or []).each do |tag| -%>
+<%= tag %>
+<% end -%>
+<% end -%>
+<% provides.each do |prov| -%>
+Provides: <%= prov %>
+<% end -%>
+<% conflicts.each do |conflict| -%>
+Conflicts: <%= conflict %>
+<% end -%>
+<% replaces.each do |repl| -%>
+<%# The closes equivalent in RPM to "replaces" is "Obsoletes" -%>
+Obsoletes: <%= repl %>
+<% end -%>
+<%# rpm rejects descriptions with blank lines (even between content), so hack
+    around it by replacing blank lines with ' .' -%>
+%description
+<%= description.gsub(/^\s*$/, " .") %>
+
+%prep
+%autosetup -n <%= source_archive_dirname %>
+
+%build
```

Comment:
The %build section was hand-written loosely based on what the cpan input already does.

I probably still need to set `-Mlocal::lib=./cpan` when invoking perl, though.

Reply:
Notably, this specfile only works with cpan input and would fail everything else.

For this to go forward, we'd to codify the prep/build/install steps for other inputs (gem, npm, pear, etc).

Reply:
I have some ideas, but we'll need to study the existing inputs (rpm, deb, gem, npm, et al) to see what common elements exist between them.